### ISSUE TITLE
email address overrides NameID

### ISSUE THREAD

**Issue by atlasntefft** [lyrasis/aspace-oauth#31]
Hello Mark,

I’ve encountered an issue where the NameID value gets replaced if I also provide an email address via urn:oid:0.9.2342.19200300.100.1.3.

In the config.rb I have 
`name_identifier_format => "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress",`

If I send over just the Unique User Identifier (Name ID) I’ll get that value made as a username.  Perfect.
The catch is that username also gets listed in the user’s Full Name and Email address fields.

To make it more user friendly, I send over urn:oid:2.16.840.1.113730.3.1.241 to take the place of Full Name.  That works great.

Where things go wrong is when I also send over urn:oid:0.9.2342.19200300.100.1.3
That's the email address and, when that's provided, the first part of that email address (eg: nathan in nathan@test.com) becomes the username.

To put it another way, it appears the email address value is getting parsed and everything before the @ gets put in as the username even if I’m sending something different as the NameID.  I can confirm that the NameID is still coming over as intended by decoding the assertion that gets recorded in the archivesspace.out file.  It's just not getting used anymore by the plugin, it seems.

For now I've asked the user to omit the email address from their assertion so that their unique NameID value gets used instead.

Thanks,
Nathan

**Comment by mark-cooper** [comment]
HI @atlasntefft,

Have you tried setting `AppConfig[:oauth_username_is_email] = true`? If you're anticipating email addresses for usernames that will prevent the plugin from manipulating the value. It owes somewhat to the legacy of ArchivesSpace disallowing an email address as the username and needing to maintain backwards compatibility now that it is possible and not wanting new users to be created (i.e. we needed something to provide opt-in or opt-out of an email address as the username -- a scenario we can handle more cleanly in the future with a mapping/transformer option on the incoming username so we won't need a dedicated, single purpose config setting for it).

**Comment by atlasntefft** [comment]
Hi @mark-cooper 

I am not attempting to use email addresses for usernames.  The issue is quite the opposite, actually.
I want to fill in the email address in the user profile but part of that value is superseding NameID when we don't want it to.

If NameID is sent over as _mcooper_ then it works fine.
If NameID is sent over as _mcooper_ and `urn:oid:0.9.2342.19200300.100.1.3` is sent over as _mark.cooper@here.com_ then the username gets created as `mark.cooper`.  We want `mcooper`.
`

**Comment by mark-cooper** [comment]
@atlasntefft I see. I'm not sure when I'll get time to sit down with it but if you could add sanitized log output here for the "assertion[s] that gets recorded in the archivesspace.out file" that may help to expedite things.

**Comment by RyanNSellars-USDA** [comment]
@atlasntefft Is this working for you now?

**Comment by mark-cooper** [comment]
Closing as a stale issue. Please create a new issue with relevant log output if wanting to reopen it.
